Add new tests for pushing to OCIR

nigeldeakin opened the pull request:
This PR adds a new test case that tests the use of `internal/docker-push` to push an image to OCIR, and the use of `internal/docker-run` to pull and run a (private) image that is on OCIR.

Example run at https://app.wercker.com/wercker/wercker-tests/runs/build/5c2f92ae0518260007ad8809?step=5c2f92c1cfa0fc00074f6995

nigeldeakin commented:
Note ot self: after this PR is merged I need to modify the branch filter for this pipeline, so that it runs on all branches (not just this temporary feature branch)

hatchan commented on wercker.yml:
```diff
@@ -391,6 +391,144 @@ test-docker-build-ecr-2:
+            fi      
+    - internal/docker-kill:
+        name: myTestContainer  
+#  after-steps:
```
Should this be commented out?

nigeldeakin replied:
Probably not. I'll uncomment it for consistency with the other pipelines.

I still haven't set the pager duty key, since some of these pipelines do occasionally (usually because the registry is unresponsive). I think now may be a good time for me to take a proper look at this.

nigeldeakin commented:
I've updated the README to list all the env vars that need to be set. 
I've updated the workflow-in-yml to include these new pipelines (and the two ECR ones)
I've tested both by setting the env vars listed in the README and then running the entire workflow locally.
